File: app/src/main/java/de/garten/training/depotflow/data/db/converter/WorkOrderStatusConverter.java
package de.garten.training.depotflow.data.db.converter;

import org.greenrobot.greendao.converter.PropertyConverter;

import de.garten.training.depotflow.domain.WorkOrderStatus;

public class WorkOrderStatusConverter implements PropertyConverter<WorkOrderStatus, String> {

    @Override
    public WorkOrderStatus convertToEntityProperty(String databaseValue) {
        return WorkOrderStatus.fromServerValue(databaseValue);
    }

    @Override
    public String convertToDatabaseValue(WorkOrderStatus entityProperty) {
        if (entityProperty == null) {
            return WorkOrderStatus.NEW.toServerValue();
        }
        return entityProperty.toServerValue();
    }
}

File: app/src/main/java/de/garten/training/depotflow/data/db/converter/WorkOrderStatusConverter.kt
package de.garten.training.depotflow.data.db.converter

import de.garten.training.depotflow.domain.WorkOrderStatus
import org.greenrobot.greendao.converter.PropertyConverter

class WorkOrderStatusConverter : PropertyConverter<WorkOrderStatus?, String?> {
    override fun convertToEntityProperty(databaseValue: String?): WorkOrderStatus? {
        return WorkOrderStatus.fromServerValue(databaseValue)
    }

    override fun convertToDatabaseValue(entityProperty: WorkOrderStatus?): String? {
        if (entityProperty == null) {
            return WorkOrderStatus.NEW.toServerValue()
        }
        return entityProperty.toServerValue()
    }
}
